### PR TITLE
[CLOUD-305] Update Azure Solution Template with more feedback

### PR DESCRIPTION
**Pull request by ryancragun**
* Move setup extension into main template
* Collapse VM subtemplates into a single resource by making `osProfile`
  data a variable.
* Use references to the `publicIpAddress` rather than the `fqdn`
  variable for GovCloud/Germany support.
* Add `artifactsBaseUrl` to `baseUrl` metadata
* Use two VM size choosers for premium and standard instances

### CODE REVIEW

**Comment by danielsdeleo** [comment]
TBH I don't know what separates good from not good here, but it makes basic sense. 👍

File: arm-templates/automate/createUiDefinition.json [hunk 1]
```diff
@@ -127,40 +160,40 @@
...
+              "Standard_DS4_v2",
+              "Standard_DS5_v2"
+            ],
             "constraints": {
-              "allowedTypes": [
-                "Standard_LRS",
-                "Standard_GRS",
-                "Premium_LRS"
+              "allowedSizes": [
+                "Standard_DS4",
```
Comment (sdelano) [inline arm-templates/automate/createUiDefinition.json]: Do we need to remove these premium allowed sizes from the standard selector?

Comment (ryancragun) [inline arm-templates/automate/createUiDefinition.json]: Nope, you can use those instance types with regular storage so I think it's okay to leave them as an option.

**Comment by sdelano** [comment]
LGTM

**Pull request merged**